<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainGeneratorWindow</class>
 <widget class="QMainWindow" name="MainGeneratorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1086</width>
    <height>782</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>CAN Tool</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1061</width>
      <height>721</height>
     </rect>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="VectorTab">
     <attribute name="title">
      <string>Vector View</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>1041</width>
        <height>671</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTableView" name="MsgTableView"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <property name="rightMargin">
          <number>500</number>
         </property>
         <item>
          <widget class="QPushButton" name="AddMsgButton">
           <property name="text">
            <string>Add Msg</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="DeleteMsgButton">
           <property name="text">
            <string>Delete Msg</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="NewBlockButton">
           <property name="text">
            <string>New Block</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="SaveBlockButton">
           <property name="text">
            <string>Save Block</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="SigTableView"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="CompactTab">
     <attribute name="title">
      <string>Compact View</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="ScriptTab">
     <attribute name="title">
      <string>Script View</string>
     </attribute>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>1031</width>
        <height>651</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="3,1">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,4,0">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Script Name</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="scriptNameLabel">
             <property name="frameShape">
              <enum>QFrame::Box</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Plain</enum>
             </property>
             <property name="lineWidth">
              <number>1</number>
             </property>
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="refreshScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/refreshIcon</normaloff>:/buttons/refreshIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>MESSAGE GROUPS</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="scriptMsgGroupList"/>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>SCRIPT SELECTION</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="scriptSelectionList"/>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="addScriptBtn">
             <property name="text">
              <string>ADD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>SCRIPT EXECUTION QUEUE</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="scriptExecutionList"/>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QCheckBox" name="playAllCheckBox">
             <property name="text">
              <string>Play All</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="clearScriptBtn">
             <property name="text">
              <string>CLEAR</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="frameShape">
              <enum>QFrame::Box</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="text">
              <string>Script Control</string>
             </property>
             <property name="margin">
              <number>11</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="playScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/playIcon</normaloff>:/buttons/playIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pauseScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/pauseIcon</normaloff>:/buttons/pauseIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="stopScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/stopIcon</normaloff>:/buttons/stopIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="prevScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/prevIcon</normaloff>:/buttons/prevIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="nextScriptBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="res.qrc">
               <normaloff>:/buttons/nextIcon</normaloff>:/buttons/nextIcon</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="flat">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1086</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoadDbcFile"/>
    <addaction name="actionLoadScript"/>
    <addaction name="actionLoad_Block"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionEditConnection"/>
    <addaction name="actionEditScript"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionViewDBCBrowser"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>RightToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoadDbcFile">
   <property name="text">
    <string>Load dbc file...</string>
   </property>
  </action>
  <action name="actionLoadScript">
   <property name="text">
    <string>Load JSON script...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionEditConnection">
   <property name="text">
    <string>Connection...</string>
   </property>
  </action>
  <action name="actionLoad_Block">
   <property name="text">
    <string>Load Block ...</string>
   </property>
  </action>
  <action name="actionEditScript">
   <property name="text">
    <string>Script...</string>
   </property>
  </action>
  <action name="actionViewDBCBrowser">
   <property name="text">
    <string>BDC browser...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>load_dbc_triggered()</slot>
 </slots>
</ui>
